src/hooks/web3/useChangeNetwork.ts

```diff
@@ -131,39 +132,103 @@ export function useChangeNetwork() {
+      })()
+
+      const errors: Error[] = []
+      for (let i = 0; i < solutionPrefer.length; i++) {
         try {
-          await activeProvider.request({
-            method: 'wallet_addEthereumChain',
-            params: [addChainParameter],
+          console.info('[Add network] start:', {
+            wallet: walletEVM.walletKey,
+            solution: solutionPrefer[i],
+            addChainParameter,
+          })
+          await solutions[solutionPrefer[i]]()
+          console.info('[Add network] success:', {
+            wallet: walletEVM.walletKey,
+            solution: solutionPrefer[i],
+            addChainParameter,
           })
           wrappedSuccessCallback()
+          return
         } catch (error) {
-          console.error('Add new network failed', { addChainParameter, error })
-          failureCallback(desiredChainId, error, customFailureCallback, customTexts)
-          if (!didUserReject(error)) {
-            const e = new Error(`[Wallet] ${error.message}`)
-            e.name = 'Add new network Error'
-            e.stack = ''
-            captureException(e, {
-              level: 'warning',
-              extra: { error, wallet: walletEVM.walletKey, chainId, addChainParameter },
-            })
+          console.error(
+            '[Add network] error:',
+            JSON.stringify(
+              {
+                wallet: walletEVM.walletKey,
+                desiredChainId,
+                solution: solutionPrefer[i],
+                message: friendlyError(error),
+                error,
+                addChainParameter,
+                didUserReject: didUserReject(error),
+              },
+              null,
+              2,
+            ),
+          )
+
+          if (didUserReject(error)) {
+            failureCallback(desiredChainId, error, customFailureCallback, customTexts)
+            return
           }
+          errors.push(error)
         }
       }
+
+      failureCallback(desiredChainId, errors.at(-1), customFailureCallback, customTexts)
+      const e = new Error(`[Add network] ${walletEVM.walletKey} ${friendlyError(errors.at(-1) || '')}`)
+      e.name = 'Add new network Error'
+      e.stack = ''
+      captureException(e, {
+        level: 'error',
+        extra: {
+          wallet: walletEVM.walletKey,
+          desiredChainId,
+          addChainParameter,
+          friendlyMessages: errors.map(friendlyError),
+          errors,
+        },
+      })
     },
-    [library?.provider, chainId, failureCallback, fetchKyberswapConfig, successCallback, walletEVM.walletKey],
+    [library?.provider, failureCallback, fetchKyberswapConfig, successCallback, walletEVM.walletKey, connector],
   )
 
   const changeNetwork = useCallback(
     async (
       desiredChainId: ChainId,
       customSuccessCallback?: () => void,
       customFailureCallback?: (error: Error) => void,
-      waitUtilUpdatedChainId = false,
+      waitUtilUpdatedChainId = false, //todo: force all to true
```

Comment:
TODO

Reply:
we might need an additional refactor on this later 😅